tests/reftests/rec-pin.test

```diff
@@ -0,0 +1,994 @@
+pin1sub2          l2         git+file://${BASEDIR}/pinnes#master  b
+pin1sub1sub2      l2         git+file://${BASEDIR}/pinnes#master  a/k
+pin1sub1sub1sub1  l2         git+file://${BASEDIR}/pinnes#master  a/i/j
+pin1sub1sub1      l2         git+file://${BASEDIR}/pinnes#master  a/i
+pin1sub1          l2         git+file://${BASEDIR}/pinnes#master  a
+pin1              l2         git+file://${BASEDIR}/pinnes#master  -
+### opam pin scan ./pinnes --subpath a | grep -v '# Name.*'
+pin1sub1  l2         git+file://${BASEDIR}/pinnes#master  a
+### opam pin scan ./pinnes --subpath a --recursive | '# Name.*' -> '\c'
+pin1sub1sub2      l2         git+file://${BASEDIR}/pinnes#master  a/k
+pin1sub1sub1sub1  l2         git+file://${BASEDIR}/pinnes#master  a/i/j
+pin1sub1sub1      l2         git+file://${BASEDIR}/pinnes#master  a/i
+pin1sub1          l2         git+file://${BASEDIR}/pinnes#master  a
+### opam pin scan ./pinnes --recursive --normalise
+pin1sub2.l2^git+file://${BASEDIR}/pinnes#master^b
+pin1sub1sub2.l2^git+file://${BASEDIR}/pinnes#master^a/k
+pin1sub1sub1sub1.l2^git+file://${BASEDIR}/pinnes#master^a/i/j
+pin1sub1sub1.l2^git+file://${BASEDIR}/pinnes#master^a/i
+pin1sub1.l2^git+file://${BASEDIR}/pinnes#master^a
+pin1.l2^git+file://${BASEDIR}/pinnes#master
+### opam pin --with-version 3 pin1sub2^git+file://${BASEDIR}/pinnes#master^b pin1sub1sub1sub1^git+file://${BASEDIR}/pinnes#master^a/i/j pin1sub1sub1^git+file://${BASEDIR}/pinnes#master^a/i pin1sub1^git+file://${BASEDIR}/pinnes#master^a pin1^git+file://${BASEDIR}/pinnes#master pin1sub1sub2^git+file://${BASEDIR}/pinnes#master^a/k -n
+This will pin the following packages: pin1sub2, pin1sub1sub1sub1, pin1sub1sub1, pin1sub1, pin1, pin1sub1sub2. Continue? [Y/n] y
+Package pin1sub2 does not exist, create as a NEW package? [Y/n] y
+[pin1sub2.3] synchronised (no changes)
+pin1sub2 is now pinned to git+file://${BASEDIR}/pinnes#master (b) (version 3)
+Package pin1sub1sub1sub1 does not exist, create as a NEW package? [Y/n] y
+[pin1sub1sub1sub1.3] synchronised (no changes)
+pin1sub1sub1sub1 is now pinned to git+file://${BASEDIR}/pinnes#master (a/i/j) (version 3)
+Package pin1sub1sub1 does not exist, create as a NEW package? [Y/n] y
+[pin1sub1sub1.3] synchronised (file://${BASEDIR}/pinnes(a/i))
+pin1sub1sub1 is now pinned to git+file://${BASEDIR}/pinnes#master (a/i) (version 3)
+Package pin1sub1 does not exist, create as a NEW package? [Y/n] y
+[pin1sub1.3] synchronised (file://${BASEDIR}/pinnes(a))
+pin1sub1 is now pinned to git+file://${BASEDIR}/pinnes#master (a) (version 3)
+Package pin1 does not exist, create as a NEW package? [Y/n] y
+[pin1.3] synchronised (file://${BASEDIR}/pinnes)
+pin1 is now pinned to git+file://${BASEDIR}/pinnes#master (version 3)
+Package pin1sub1sub2 does not exist, create as a NEW package? [Y/n] y
+[pin1sub1sub2.3] synchronised (file://${BASEDIR}/pinnes(a/k))
+pin1sub1sub2 is now pinned to git+file://${BASEDIR}/pinnes#master (a/k) (version 3)
```

Comment:
I was under the impression that the main feature of subpath was to be able to build/install opam projects located in subdirectories. However I’m not seeing any tests for the building part (most crucial I would say).

Such simple tests could something like:
```
structure:
* pinnes/
  - some-root-file
  - a
    - test.opam
    - test.install
    - some-sub-file
## a/test.opam
opam-version: "2.0"
build: ["ls"]
## a/test.install
lib: ["some-sub-file"]
### opam pin -n pinnes/ --subpath a
[pinning test]
### opam install --verbose test
[should show only test.opam, test.install and some-sub-file]
### opam list --installed-files
lib/test/some-sub-file
```

Reply:
we can add one yes